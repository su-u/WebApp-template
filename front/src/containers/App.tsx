import React from 'react';
import { NavBar } from '@/components/header/NavBar';

export const App: React.FC = () => {
  return (
    <>
      <NavBar title="Home" />
      <div data-test="div1">テストアプリテキスト</div>
      <div>divテキスト</div>
      <h2>見出しテキスト</h2>
    </>
  );
};
