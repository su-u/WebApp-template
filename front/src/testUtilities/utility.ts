export const sel = (id: string) => `[data-test="${id}"]`;
