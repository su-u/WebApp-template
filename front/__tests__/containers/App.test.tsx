import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '@/containers/App';
import { sel } from '@/testUtilities/utility';

describe('App', () => {
  test('SnapShot', () => {
    const root = renderer.create(<App />).toJSON();
    expect(root).toMatchSnapshot();
  });
  test('div1', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(sel('div1')).text()).toEqual('テストアプリテキスト');
  });
  test('h2', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('h2').text()).toEqual('見出しテキスト');
  });
});
