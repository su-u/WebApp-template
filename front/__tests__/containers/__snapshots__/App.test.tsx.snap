// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App SnapShot 1`] = `
Array [
  <div
    className="sc-AykKD fdMVYA"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary sc-AykKE gRccnS mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root sc-AykKG hRKpiD"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root sc-AykKF hLiSe MuiTypography-h6"
        >
          Home
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text sc-AykKC bGZJKL"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
      </div>
    </header>
  </div>,
  <div
    data-test="div1"
  >
    テストアプリテキスト
  </div>,
  <div>
    divテキスト
  </div>,
  <h2>
    見出しテキスト
  </h2>,
]
`;
